### PR TITLE
Documentation suggestions for spark streaming.

### PR DESCRIPTION
**Pull request by ScrapCodes**

### CODE REVIEW

File: streaming/src/main/scala/spark/streaming/receivers/ActorReceiver.scala [hunk 1]
```diff
@@ -27,12 +27,13 @@ object ReceiverSupervisorStrategy {
...
+ * Find more details at: http://spark-project.org/docs/latest/streaming-custom-receivers.html
  *
  * @example {{{
- * 	class MyActor extends Actor with Receiver{
- * 		def receive {
- * 			case anything :String ⇒ pushBlock(anything)
- * 		}
+ *  class MyActor extends Actor with Receiver{
+ *      def receive {
+ *          case anything :String ⇒ pushBlock(anything)
```
Comment (tdas) [inline streaming/src/main/scala/spark/streaming/receivers/ActorReceiver.scala]: Why was the formatting changed here? Even though its in the comments (so matters less), It would still be better to match the formatting of the actual code (that is, 2 spaces for indentation).

File: docs/streaming-custom-receivers.md [hunk 2]
```diff
@@ -7,10 +7,45 @@ A "Spark Streaming" receiver can be a simple network stream, streams of messages
...
+         protected def onStop() {
+           blocksGenerator.stop()
+         }
+
+       }
+
+{% endhighlight %}
+
+
+All we did here is extended NetworkReceiver and called blockManager api method (i.e. +=) to push our blocks of data. Please refer to scala-docs of NetworkReceiver for more details.
```
Comment (tdas) [inline docs/streaming-custom-receivers.md]: += is blockGenerators API, not block managers. Also api -> API. And the blockGenerator does the job of dividing the received data into blocks.

On a related note, a different thought occurred to me. The whole blockGenerator deal can be made much simpler. It could be that the blockGenerator is defined in the NetworkReceiver class for everyone to use, and calls to add data (that is, the += ) can be simply wrapped in a function NetworkReceiver.pushData. Then all the user has to do call pushData and not worry about creating a blockGenerator. What do you think?

**Comment by tdas** [comment]
Jenkins, please test this.

**Comment by tdas** [comment]
@ScrapCodes Overall, this looks good to me. There were a few minor comments and thoughts.

**Comment by tdas** [comment]
@ScrapCodes Overall, this looks good to me. There were a few minor comments and thoughts.

File: docs/streaming-custom-receivers.md (hunk 2 above)
Comment (ScrapCodes) [inline docs/streaming-custom-receivers.md]: Your thoughts regarding pushData function are good and infact two can alias each other. Let's do that in a different pull request and let this just be a documentation patch.

Comment (tdas) [inline docs/streaming-custom-receivers.md]: Yes definitely. I didnt mean it to be for this pull request.
In fact, I have some important optimization I have made to block generator that helps in high throughput data ingestion that touches most of the code in that area, so I will take care of it when I work on it.

**Comment by ScrapCodes** [comment]
@tdas Do you think, I have completely incorporated your suggestions ? Or did I miss something.

**Comment by andyk** [comment]
Sadly I think Jenkins is picky about the ordering of words when you ask...

Jenkins test this please

**Comment by andyk** [comment]
Hmm, let's try this...

Jenkins, ok to test.

**Comment by andyk** [comment]
Sorry for the spam. I'm debugging the Jenkins pull request builder plugin. I made a change that might be a workaround to bug. Trying to trigger a Jenkins build again...

Jenkins, test this please.

**Comment by andyk** [comment]
Jenkins test this please

**Comment by mateiz** [comment]
@tdas is this good to commit?

File: docs/streaming-programming-guide.md [hunk 3]
```diff
@@ -301,6 +301,9 @@ dstream.checkpoint(checkpointInterval) // checkpointInterval must be a multiple
 For DStreams that must be checkpointed (that is, DStreams created by `updateStateByKey` and `reduceByKeyAndWindow` with inverse function), the checkpoint interval of the DStream is by default set to a multiple of the DStream's sliding interval such that its at least 10 seconds.
 
 
+## Customizing Receiver
+Spark comes with a built in support for most common usage scenarios where input stream source can be either a network socket stream to support for a few message queues. Apart from that it is also possible to supply your own custom receiver via a convenient API. Find more details at [Custom Receiver Guide](streaming-custom-receivers.md)
```
Comment (tdas) [inline docs/streaming-programming-guide.md]: Should the link be to streaming-custom-receivers.md or streaming-custom-receivers.html ? If the markdown automatically converts it then ignore this.

**Comment by tdas** [comment]
Other than the potential linking problem that I commented on, this is good to commit. Sorry I totally forgot about this pull request.

**Comment by mateiz** [comment]
Yeah, good point. That should link to .html.

**Comment by ScrapCodes** [comment]
Hey Matei and TD, I have done a forced push to clean the commits and also incorporate review feedback.

**Comment by mateiz** [comment]
Cool, thanks! I've merged it in.
